src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/TableLayoutPanelDesigner.cs

```diff
@@ -0,0 +1,2420 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System.Collections;
+using System.ComponentModel;
+using System.ComponentModel.Design;
+using System.Drawing.Design;
+using System.Drawing;
+using System.Windows.Forms.Design.Behavior;
+using System.CodeDom;
+using System.ComponentModel.Design.Serialization;
+using System.Text.RegularExpressions;
+
+namespace System.Windows.Forms.Design;
+
+internal class TableLayoutPanelDesigner : FlowPanelDesigner
+{
+    private TableLayoutPanelBehavior tlpBehavior;//every resize col/row glyph is associated with this instance of behavior
+    private Point droppedCellPosition = InvalidPoint;//used to insert new children
+
+    // NEVER USE undoing DIRECTLY. ALWAYS USE THE PROPERTY
+    private bool undoing;
+    private UndoEngine undoEngine;
+
+    private Control localDragControl;//only valid if we're currently dragging a child control of the table
+    private ArrayList dragComps;          //the components we are dragging
+    private DesignerVerbCollection verbs;//add col/row and remove col/row tab verbs
+    private DesignerTableLayoutControlCollection controls;
+    private DesignerVerb removeRowVerb;
+    private DesignerVerb removeColVerb;
+    private DesignerActionListCollection actionLists;//action list for the Smart Tag
+
+    private BaseContextMenuStrip designerContextMenuStrip;
+    private int curRow = -1; //row cursor was over when context menu was dropped
+    private int curCol = -1;  //col cursor was over when context menu was dropped
+
+    private IComponentChangeService compSvc;
+    private PropertyDescriptor rowStyleProp;
+    private PropertyDescriptor colStyleProp;
+
+    // Only used when adding controls via the toolbox
+    private int rowCountBeforeAdd; // What's the row count before a control is added
+    private int colCountBeforeAdd; // Ditto for column
+
+    // TLP context menu row/column items.
+    private ToolStripMenuItem contextMenuRow;
+    private ToolStripMenuItem contextMenuCol;
+
+    private int ensureSuspendCount;
+
+    private TableLayoutPanelBehavior Behavior
+    {
+        get
+        {
+            tlpBehavior ??= new TableLayoutPanelBehavior(Table, this, Component.Site);
+
+            return tlpBehavior;
+        }
+    }
+
+    private TableLayoutColumnStyleCollection ColumnStyles
+    {
+        get
+        {
+            return Table.ColumnStyles;
+        }
+    }
```

Comment:
I see there are still some properties that can be changed to expression-body style, but I'm fine with cleaning-up in the next PR
```suggestion
    private TableLayoutColumnStyleCollection ColumnStyles => Table.ColumnStyles;
```

Reply:
Done.